app-near/src/server/carbon-footprint/convert.ts

```diff
@@ -135,15 +147,173 @@ export const convertCarbonFootprintBody = (
       body.calculatedResults["transport . mobilité douce"],
     transportationTrain: body.calculatedResults["transport . train"],
 
-    broadcastChannel: body.answers.broadcastChannel,
-    knowSu: body.answers.knowSu,
-    su: body.answers.su, // near-47 to confirm we can get it from body
-    neighborhood: body.answers.neighborhood,
-    airTravelFrequency: body.answers.airTravelFrequency,
-    digitalIntensity: body.answers.digitalIntensity,
-    heatSource: body.answers.heatSource,
-    meatFrequency: body.answers.meatFrequency,
-    purchasingStrategy: body.answers.purchasingStrategy,
-    transportationMode: body.answers.transportationMode,
+    neighborhood: body.neighborhoodId,
+    broadcastChannel: body.broadcastChannel,
+
+    knowSu: mapConnaissanceSu(
+      body.answers.userAnswers["services sociétaux . connaissance su"],
+    ),
+    su: mapSu(body.answers.userAnswers["services sociétaux . su . choix"]),
+
+    airTravelFrequency: mapAirTravelFrequency(
+      body.answers.userAnswers[
+        "services sociétaux . su . calcul su . question 5"
+      ],
+    ),
+
+    heatSource: mapHeatSource(
+      body.answers.userAnswers[
+        "services sociétaux . su . calcul su . question 6"
+      ],
+    ),
+    meatFrequency: mapMeatFrequency(
+      body.answers.userAnswers[
+        "services sociétaux . su . calcul su . question 1"
+      ],
+    ),
+
+    transportationMode: mapTransportationMode(
+      body.answers.userAnswers[
+        "services sociétaux . su . calcul su . question 2"
+      ],
+    ),
+    digitalIntensity: mapDigitalIntensity(
+      body.answers.userAnswers[
+        "services sociétaux . su . calcul su . question 3"
+      ],
+    ),
+    purchasingStrategy: mapPurchasingStrategy(
+      body.answers.userAnswers[
+        "services sociétaux . su . calcul su . question 4"
+      ],
+    ),
   };
 };
+
+export const ConnaissanceSuEnumInput = {
+  oui: true,
+  non: false,
+};
+
+export type ConnaissanceSuEnumKey = keyof typeof ConnaissanceSuEnumInput;
+
+const mapConnaissanceSu = (key: string | undefined): boolean => {
+  return !!key && key in ConnaissanceSuEnumInput
+    ? ConnaissanceSuEnumInput[key as keyof typeof ConnaissanceSuEnumInput]
+    : false;
+};
```

Comment:
:green_circle:  ça me parai compliqué, pourquoi ne pas avoir simplement fait ça ?

```suggestion
export enum ConnaissanceSuEnumInput {
  YES =  "oui",
  NO = "non",
};

const mapConnaissanceSu: Record<ConnaissanceSuEnumInput, boolean> =  {
 [ConnaissanceSuEnumInput.YES]: true,
 [ConnaissanceSuEnumInput.NO]: false,
 };
```

et ensuite utiliser le mapper directement, exemple : 

```
knowSu: mapConnaissanceSu[body.answers.userAnswers["services sociétaux . connaissance su"] ?? false,
```

pareil pour les autres

Reply:
Ca produit une erreur typescript Element implicitly has an 'any' type because expression of type 'any' can't be used to index type 'Record<ConnaissanceSuEnumInput, boolean>'.ts(7053)

Reply:
J'aime pas du tout l'implémentation actuelle non plus